listenbrainz/webserver/views/api_tools.py

```diff
@@ -326,15 +325,16 @@ def publish_data_to_queue(data, exchange, queue, error_msg):
         error_msg (str): the error message to be returned in case of an error
     """
     try:
-        with rabbitmq_connection._rabbitmq.acquire() as cxn:
-            cxn.channel.exchange_declare(exchange=exchange, exchange_type='fanout')
-            cxn.channel.queue_declare(queue, durable=True)
-            cxn.channel.basic_publish(
-                exchange=exchange,
-                routing_key='',
-                body=ujson.dumps(data),
-                properties=pika.BasicProperties(delivery_mode=2, ),
-            )
+        cxn, channel = rabbitmq_connection._rabbitmq.get()
```

Comment:
I gotta say, I like the concept of the with ... as: statement here. It isn't hard to implement -- do you think we should do this?